mdsuite/database/simulation_database.py

```diff
@@ -210,47 +209,27 @@ class Database:
 
     Attributes
     ----------
-    architecture : str
-                The type of the database_path implemented, either a simulation
-                database_path, or an analysis database_path.
-
-    name : str
+    path : str|Path
             The name of the database_path in question.
     """
 
-    def __init__(self, architecture: str = "simulation", name: str = "database"):
+    def __init__(self, path: typing.Union[str, pathlib.Path] = "database"):
         """
         Constructor for the database_path class.
 
         Parameters
         ----------
-        architecture : str
-                The type of the database_path implemented, either a simulation
-                database_path, or an analysis database_path.
-        name : str
+        path : str|Path
                 The name of the database_path in question.
         """
-        # TODO support pathlib
-
-        self.architecture = architecture  # architecture of database_path
-        self.name = name  # name of the database_path
-
-    @staticmethod
-    def close(database: hf.File):
-        """
-        Close the database_path
-
-        Parameters
-        ----------
-        database : hf.File
-                Database to close
-
-        Returns
-        -------
-        Closes the database_path object
-        """
-
-        database.close()
+        if isinstance(path, pathlib.Path):
+            self.path = path.as_posix()
+        elif isinstance(path, str):
+            self.path = path  # name of the database_path
+        else:
+            # TODO fix this!
```

Comment:
what fix are you looking for?

Reply:
Related to #453 this should only take str or pathlib and not <class byte>

Reply:
ahh okay.